### ISSUE TITLE
PETSc wrappers lack SuperLU-Dist direct solver

### ISSUE THREAD

**Issue by davydden** [dealii/dealii#3268]
subj.

**Comment by davydden** [comment]
when implemented, we will need `DEAL_II_PETSC_WITH_SUPERLU_DIST` in https://github.com/dealii/dealii/blob/master/cmake/configure/configure_3_petsc.cmake

**Comment by jppelteret** [comment]
I don't know how `PETSc` interfaces to solvers, but it would be great if one could switch between the direct solvers in the same manner that [one can do with Trilinos](http://www.dealii.org/developer/doxygen/deal.II/structTrilinosWrappers_1_1SolverDirect_1_1AdditionalData.html) (namely with via a string). This would make integrating them into the `generic_linear_algebra` module easier.

**Comment by davydden** [comment]
>  it would be great if one could switch between the direct solvers in the same manner that one can do with Trilinos (namely with via a string).

agree, that should be possible. It happens around these lines: https://github.com/dealii/dealii/blob/master/source/lac/petsc_solver.cc#L774-L778

**Comment by davydden** [comment]
while we are here, there should also be `SolverDirect::initialize(const SparseMatrix & A)` (similar to [Trilinos direct solver](https://www.dealii.org/developer/doxygen/deal.II/classTrilinosWrappers_1_1SolverDirect.html#a6c0fecc921bc93f2c005e265cfbdd882)) to do factorization once and then have an option to apply it to different RHS.
